spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -1226,12 +1227,51 @@ private void rollback(long deliveryTag, Exception e) {
 			}
 		}
 
-		protected synchronized void sendAck(long now) throws IOException {
-			getChannel().basicAck(this.latestDeferredDeliveryTag, true);
+		protected synchronized void sendAck(long now) throws Exception {
+			sendAckWithNotify(this.latestDeferredDeliveryTag, true);
 			this.lastAck = now;
 			this.pendingAcks = 0;
 		}
 
+		/**
+		 * Send ack and notify MessageAckListener(if set).
+		 * @param deliveryTag DeliveryTag of this ack.
+		 * @param multiple Whether multiple ack.
+		 * @throws Exception Occured when ack.
+		 */
+		private void sendAckWithNotify(long deliveryTag, boolean multiple) throws Exception {
+			try {
+				getChannel().basicAck(deliveryTag, multiple);
+				notifyMessageAckListener(getMessageAckListener(), true, deliveryTag, null);
+			}
+			catch (Exception e) {
+				notifyMessageAckListener(getMessageAckListener(), false, deliveryTag, e);
+				throw e;
+			}
+		}
+
+		/**
+		 * Notify MessageAckListener set on message listener.
+		 * @param messageAckListener MessageAckListener set on the message listener.
+		 * @param success Whether ack succeeded.
+		 * @param deliveryTag The deliveryTag of ack.
+		 * @param cause If an exception occurs.
+		 */
+		private void notifyMessageAckListener(@Nullable MessageAckListener messageAckListener,
+											  boolean success,
+											  long deliveryTag,
+											  @Nullable Throwable cause) {
+			if (messageAckListener == null) {
```

Comment:
We can avoid this null check by making the default `(success, tag, ex) -> { }` and `Assert.notNull(...)` in the setter.